plugins/router/f5/f5.go

```diff
@@ -0,0 +1,555 @@
+package f5
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/golang/glog"
+)
+
+type F5Error struct {
+	verb           string
+	url            string
+	httpStatusCode int
+	err            error
+	JsonStatusCode int     `json:"code"`
+	Message        *string `json:"message"`
+}
+
+func (err F5Error) Error() string {
+	var msg string
+
+	if err.err != nil {
+		msg = fmt.Sprintf("error: %v", err.err)
+	} else if err.Message != nil {
+		msg = fmt.Sprintf("HTTP code: %d; error from F5: %s",
+			err.httpStatusCode, *err.Message)
+	} else {
+		msg = fmt.Sprintf("HTTP code: %d.", err.httpStatusCode)
+	}
+
+	return fmt.Sprintf("Encountered an error on %s request to URL %s: %s",
+		err.verb, err.url, msg)
+}
+
+type f5LTM struct {
+	host     string
```

Comment:
I did it this way to parallel `templateRouter` and `templateRouterCfg`, although there `templateRouterCfg` is not strictly a subset of `templateRouter`, so following the same approach is misguided.  I'll follow your suggestion here.